### PR TITLE
Add multiple emoloyee in salary structure

### PR DESCRIPTION
**Pull request by mainul94**

### CODE REVIEW

**Comment by rmehta** [comment]
@mainul94 thanks for the contribution

Can you add some screenshots for this?

Some other thoughts: Maybe we can skip building. Directly filter and check employees from the list view and then set salary structure?

**Comment by mainul94** [comment]
@rmehta 
I added a button ("Sync employee") in salary structure and redirect to a single DocType ("Salary Structure Employee Assign") with salary structure Id and Company.
In "Salary Structure Employee Assign" DocType user can see also Employee who Belong this Salary Structure and who not.
As Link Attendance, User can add or remove Employee by Check or Uncheck.
salary_structure
employee_assign

**Comment by rmehta** [comment]
@mainul94 I still think this will be an overkill to maintain a separate UI for this. How about we add a simple dialog inside the Salary Structure and ask users to work with just a couple of filters, maybe company, department, branch?

**Comment by mainul94** [comment]
@rmehta 
Ok I will fix and add dialog for add Employee.

**Comment by mainul94** [comment]
@rmehta Now Multi Employee add in a Dialog.

**Comment by rmehta** [comment]
👍 great! I would usually not go with so much code. 

Maybe just add it via the filters and let users delete from the table itself?

**Comment by mainul94** [comment]
@rmehta Yes user can delete from table itself. It's validated if any employee in exists in table then not add again.

File: erpnext/hr/doctype/salary_structure/salary_structure.py [hunk 1]
```diff
@@ -58,4 +58,9 @@ def postprocess(source, target):
...
 		return frappe.get_print(doc.doctype, doc.name, doc = doc, print_format = print_format)
 	else:
-		return doc
\ No newline at end of file
+		return doc
+
+
+@frappe.whitelist()
+def get_employees(**args):
+	return frappe.get_all('Employee',filters=args['filters'], fields=['name', 'employee_name'])
```
Comment (rmehta) [inline erpnext/hr/doctype/salary_structure/salary_structure.py]: use `frappe.get_list` here. Permissions matter!

File: erpnext/hr/doctype/salary_structure/salary_structure.js [hunk 2]
```diff
@@ -32,14 +32,120 @@ frappe.ui.form.on('Salary Structure', {
...
+			callback:function (r) {
+				frm.$empDialog.__data = r.message;
+				frm.trigger('setEmployee')
+			}
+		})
+	},
+	
+	
+	setEmployeesInChildTable:function (frm) {
+		var employees = $.map(frm.doc.employees, function(d) { return d.employee });
```
Comment (rmehta) [inline erpnext/hr/doctype/salary_structure/salary_structure.js]: Remove the checkbox code, just add the employees in the table and then let the user decide

File: erpnext/hr/doctype/salary_structure/salary_structure.js [hunk 3]
```diff
@@ -32,14 +32,120 @@ frappe.ui.form.on('Salary Structure', {
...
+			frm.$empDialog.__check_all = false;
+			frm.trigger('employeeCheckToggle');
+		});
+		frm.$empDialog.get_input('set_employee').addClass('btn-primary').on('click', function () {
+			frm.trigger('setEmployeesInChildTable');
+		});
+		frm.$empDialog.show();
+	},
+
+	getEmployees:function (frm) {
```
Comment (rmehta) [inline erpnext/hr/doctype/salary_structure/salary_structure.js]: We usually use `snake_case`, so for consistency sake, `get_employees` will be better :)

**Review by rmehta — CHANGES_REQUESTED** [review]

**Comment by rmehta** [comment]
Let the user delete the employees from the table itself, even lesser code :)

**Comment by mainul94** [comment]
@rmehta Please check it now.

File: erpnext/hr/doctype/salary_structure/salary_structure.js [hunk 4]
```diff
@@ -32,14 +32,71 @@ frappe.ui.form.on('Salary Structure', {
...
+				{fieldname:'base_variable', fieldtype:'Section Break'},
+				{fieldname:'base', fieldtype:'Currency', label: __('Base')},
+				{fieldname:'base_col_br', fieldtype:'Column Break'},
+				{fieldname:'variable', fieldtype:'Currency', label: __('Variable')},
+				{fieldname:'check_sec', fieldtype:'Section Break'},
+				{fieldname:'set_employee', fieldtype:'Button', label: __('Add')}
+			]
+		});
+
+		frm.$emp_dialog.get_input('set_employee').removeClass('btn-xs').addClass('btn-primary pull-right').on('click', function () {
```
Comment (mainul94) [inline erpnext/hr/doctype/salary_structure/salary_structure.js]: Fixed Please Check.

**Comment by rmehta** [comment]
:+1: awesome :)

**Pull request merged**